ImageGallery/ImageGallery/ImageStore 2.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  DownloadStore.swift
+//  ImagaeGallery
+//
+//  Created by Nick Ryan on 6/18/24.
+//
+
+import SwiftUI
+import Combine
+import Foundation
+
+@MainActor
```

Comment:
Avoid marking whole objects `@MainActor`. This can cause some headaches later and in unit testing. Mark just the methods that change any values that should be observed in the main thread.